torch/csrc/jit/backends/backend_detail.cpp

```diff
@@ -63,6 +64,11 @@ Module codegen_backend_module(
       std::make_shared<CompilationUnit>(),
       /*shouldMangle=*/true);
 
+  // 1. Initialized debug infor recorder with loweredModule pointer.
+  // 2. Later call debug_info_recorder.stopRecording() to gather
+  //    recorded debug info and save it in a static global map.
+  BackendModuleDebugInfoRecorder debug_info_recorder(loweredModule._ivalue());
```

Comment:
It's a bit odd that you need to call `stopRecording` but not `startRecording`. It's as if the class is but also is not RAII.

Reply:
Yeah. It is quantum superposition or whatever it is called. ;).
Reason I did not do RAII is because, addDebugInfoMap, which can throw if debug info map for the said module already exist. And throwing inside dtor is not quite safe.
If it makes it cleaner I can add startRecording. Or any other suggestion is welcome. I did not quite like having to do stopRecording.

Reply:
As per the new design I commented above `addDebugInfoMap` can be moved to new `LoweredModule` class.